refactor(components): remove addons-bluemix components

joshblack opened the pull request:
Depends on: carbon-design-system/carbon-addons-bluemix#3

Closes: Bluemix/carbon-issues#364 (internal)

`carbon-components` contains deprecation notices for the following components: 

- Cards
- Detail Page Header
- Interior Left Nav
- Modules
- Order Summary

#### Changelog

**New**

**Changed**

- Deprecation notices for the following:
  - Cards
  - Detail Page Header
  - Interior Left Nav
  - Modules
  - Order Summary

**Removed**

joshblack commented:
A note for reviewers, would it be better to just add in a deprecation notice instead of removing these before the v8 release?

asudoh commented:
@joshblack, Thank you for doing this! Please make sure `localhost:3000` with `npm run dev` still work - You may need (circular) dependency to the addons repo.

asudoh commented:
Oh BTW I'd vote for deprecation notice, e.g. with `console.warn()` - People have complained against breaking changes with minor release update. Thanks!

joshblack commented:
Cool! Is the `warning` module brought in already? Would love to use that. To follow-up, for deprecation notices what's standard for a SCSS module and a JS module? i.e. where should I put them, what kind of message format should it have, etc.

tw15egan commented:
Yeah, we'll probably need to add in a deprecated warning until our next breaking release. Are we going to bundle this in with V8? 

As far as standards, I don't think we've formalized anything like that. We have a few mixins that have been deprecated that just tell you that it will eventually be removed. Open to suggestions in this area

joshblack commented:
I think for the JS-based files, we could add the warning in the constructor so that way it corresponds with when someone uses it (versus it just being required), for the scss stuff I'm unsure how to do that since it seems like it would just give a warning for everyone regardless of if they're actually using the component (unless they are manually specifying their scss imports).

Regardless, I'll update the PR to not remove the files and will start adding in the deprecation notices for the JS-based component files. I'll try and borrow patterns from how React communicates deprecations.

joshblack commented:
For the `deprecate` mixin, am just logging out:

```scss
WARNING: Deprecated code was found, it should be removed before the next release of Carbon.
REASON: Accessing the `card` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
```

Usage looks like:

```scss
@include exports('card') {
  @include deprecate(
    'Accessing the `card` component from the `carbon-components` package is ' +
    'deprecated. Use the `carbon-addons-bluemix` package instead.'
  ) {
    .bx--card {
      @include reset;
      @include helvetica;
      @include layer('raised');
      // ...
    }
  }
}
```

Commit:
chore(scss): add deprecation notices

asudoh commented on src/components/card/card.js:
```diff
@@ -17,6 +20,14 @@ class Card extends mixin(createComponent, initComponentBySearch) {
     this.element.addEventListener('keydown', event => {
       this._cardKeyPress(event);
     });
+    // if (__DEV__) {
```
@joshblack Looks that you are experimenting `warning` NPM module. Is it correct? If so - Please ping me if you finished with this - Thanks!

joshblack replied:
Hi @asudoh! Yeah was planning on tackling that today, although it seems like there isn't a "DEV" mode in carbon-components right now to communicate these kinds of warnings. Is that correct? Would hate to include `warning` or `console.error` or something and have it ship to prod.

Commit:
fix: revert modal change for module

Commit:
chore: add deprecation notices to JS modules

Commit:
chore: update deprecation notices

Brings in the following to help deal with deprecations:

- a babel plugin to transform __DEV__ identifiers to
  `process.env.NODE_ENV` checks.
- a rollup plugin to help inline the `NODE_ENV` value so `warning` is
  not shipped in rollup builds

joshblack commented:
Just pushed up a commit that helps handle the `__DEV__` expression calls like we were talking about 😄 

Now, when using the JS modules in `development`, the following should be logged:

<img width="589" alt="screen shot 2017-10-25 at 11 36 24 am" src="https://user-images.githubusercontent.com/3901764/32011056-43f9e9ee-b979-11e7-847d-ce17cebf93b5.png">

In addition, when compiling the CSS (from the sass entrypoint) we get:

```
WARNING: Deprecated code was found, this code will be removed before the next release of Carbon.
REASON: Accessing the `card` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
REASON: Accessing the `detail-page-header` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
REASON: Accessing the `module` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
REASON: Accessing the `interior-left-nav` component from the `carbon-components`package is deprecated. Use the `carbon-addons-bluemix` package instead.
REASON: Accessing the `order-summary` component from the `carbon-components` package is deprecated. Use the `carbon-addons-bluemix` package instead.
```

Commit:
chore: update eslint to include __DEV__

Commit:
chore: update karma tests to use new rollup config

asudoh commented on src/components/modal/_modal.scss:
```diff
@@ -16,7 +16,7 @@
   .bx--modal {
     @include reset;
-    @include font-family;
+    @include helvetica;
```
@joshblack Did you really mean to make changes here?

joshblack replied:
Nope! Thanks for catching 😄

Commit:
fix: revert change from merge

tw15egan reviewed (approved):
Looks good to me! Modules are warning as expected

joshblack commented:
Figured I'd ask while we wait for another review, did we want to ship this in a minor or major release?

tw15egan commented:
We'll need to do a major bump for 8, should this one be it? (Even though it's not breaking anything)

marijohannessen commented:
👍 ✅ 

I think this should be a major release, maybe in the same one you can switch `css--plex` to `true`? I'm going to release the new website tonight around 10pm, and I need the v8 version of the components before that so do you mind getting the commit message ready for a major release so I can merge this around 9:30pm tonight? Thanks!

asudoh reviewed (approved):
Thank you for all the effort @joshblack put here! LGTM assuming that this goes only to our major release that @marijohannessen mentioned.

marijohannessen committed:
Merge branch 'master' into 364-remove-addons-components

marijohannessen commented:
Opened this PR up in a new PR with the Plex changes
